Address issue #63: voter services with different contacts

cjerdonek opened the pull request:
This is a first crack at addressing issue #63.

It addresses both of the following use cases: 1) the use case raised by @jungshadow of two different locations / departments handling different services, and 2) the use case mentioned by myself of different services having their own contact information within a department.

It also stubs out a `VoterServiceType` enum to address the common types of services.

cjerdonek commented:
PR updated (rebased to latest from the vip5 branch).

jdmgoogle commented on vip_spec.xsd:
```diff
@@ -126,12 +124,32 @@ reference elements for polling location
-              <!-- Note: The "Hours" element is being deprecated and will be removed
-                   in future versions of VIP.
-                -->
-              <xs:element name="Hours" type="InternationalizedText" minOccurs="0" />
```
Could you add the Hours and HoursOpenId field to the Department object, too?

jdmgoogle replied:
Actually, nevermind. I just remembered that the Contact data has HoursOpenId in it.

jdmgoogle commented on vip_spec.xsd:
```diff
@@ -630,6 +652,14 @@ reference elements for polling location
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="absentee-ballots" />
+      <xs:enumeration value="polling-places" />
+      <xs:enumeration value="voter-registration" />
```
Make sure to put an "other" type in here. We should only check the "OtherType" field if "other" is present in the "Type" field.

jungshadow commented on vip_spec.xsd:
```diff
@@ -113,12 +113,10 @@ reference elements for polling location
-              <!-- Q: Please remind me what OVC stands for so we can expand this. -->
+              <!-- Q: Please remind me what OVC stands for so we can expand this,
+                      and should we move it to Department?
+              -->
```
**OVC:** Overseas Voting Contact, if there's a particular person that handles said requests

jungshadow replied:
Regarding the second part of the question, you may not need to if `ElectionAdministration.Department` is unbounded. A single `ContactId` per `ElectionAdministration.Department` may serve that purpose if the departments are individually defined.

jdmgoogle commented:
Overall this looks pretty good. Comments, @jungshadow @jktomer @pstenbjorn @nomadaisy @pkoms?

jungshadow commented:
This looks solid.

cjerdonek commented:
Thanks, guys.  I updated the PR.  For the "OvcId", I went ahead and added "overseas-voting" to the VoterServiceType enumeration, and then added an ElectionOfficialPersonId to the VoterService type.  This lets one optionally designate a person for a voter service, in addition to contact info.

jungshadow commented:
:+1:

jungshadow commented:
@cjerdonek Are you comfortable with deleting the branch?

cjerdonek commented:
Done.
